### ISSUE TITLE
As a CIF coordinator, I want to see individual milestones in the TaskList, so that I can see each milestone's individual status and navigate to each easily

### ISSUE THREAD

**Issue by dleard** [bcgov/cas-cif#623]
#### Description:

The TaskList should show an overall status for the Milstones (if a milestone is late, it would show late. If the milestones are on track it would show on track).
Each milstone should also be listed in the TaskList individually with a status and a link that will direct the user to the milestones page and directly to that milestone using an anchor.

Wireframe: https://www.figma.com/file/PzOY8RyQnVXnjZJdSv2AqQ/Wireframing_Figma?node-id=1209%3A2663

#### Acceptance Criteria:

Given I am viewing the TaskList
When there are multiple milestones
Then I can see all milestones with links in the TaskList

Given I am viewing the TaskList
When a milestone is late
Then I can see an overall status of late

Given I am viewing the TaskList
When milestones are on track
Then I can see an overall status of on track

Given I am viewing the TaskList
When I click on a milestone link
Then I am taken directly to that milestone in the milestone page

#### Development Checklist:

- [x] An overall status for milestones shows whether they are on track or late
- [x] All existing milestones are shown with links in the TaskList
- [x] Clicking a milestone in the TaskList takes the user directly to that milesone
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [x] User story is included
- [ ] User role and type are identified
- [x] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

-

**Comment by pbastia** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Climate-Action-Secretariat-60ca4121764d710011481ca2/issues/bcgov/cas-cif/623?planning-poker) with ZenHub @BCerki @dleard @gurjmatharu @matthieu-foucault @mikevespi @Sepehr-Sobhani

**Comment by pbastia** [comment]
@suhafa  does it mean we have one milestone by default? Or does the link become "Milestone reports" in the tasklist, when there are no milestones available?

**Comment by dleard** [comment]
We decided earlier this week that we don't want default miletstones.

**Comment by suhafa** [comment]
@pbastia @dleard The design has one (empty) milestone by default. Is there a reason why we decided to not have default milestones?

**Comment by dleard** [comment]
At project creation time, milestones are probably not going to be set right away, so having one by default means attempting to submit the project will be blocked on validation errors until it is removed. Some projects may never even have milestones to enter (projects that do not get approved for funding). Quarterly and Annual reports are not pre-populated with a default report.

**Comment by pbastia** [comment]
Maybe instead we can have a tasklist item that just says "Milestone reports" to mirror what we have for the other reports, and that link would be replaced by the first milestone when it is created?

**Comment by suhafa** [comment]
Good point @dleard didn't realize that constraint. In usability testing sessions, the CIF team clicked on the "Milestone report" many times, and other task list headings to go to that particular page. I wonder if we can keep it clickable even when it has sub-items, especially in the case of forms (quarterly and annual reports too) since all forms are in one page any way (not in the case of Project Overview perhaps where it doesn't have a page of its own, and only has sub-task list items as of now, which could again change in the future)

**Comment by pbastia** [comment]
@suhafa the behaviour of those headers is to expand when needed, to avoid having a large task list when most items have been left untouched. If we changed that behaviour, we'd lose the functionality
